### PR TITLE
[DMS-116] Add Schoolyeartypes e2e features

### PR DESCRIPTION
**Pull request by CSR2017**

### CODE REVIEW

File: src/tests/EdFi.DataManagementService.Tests.E2E/Features/SchoolYearReferenceValidation.feature [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                  """
+                  {
+                      "calendarCode": "2010605675",
+                      "schoolReference": {
+                        "schoolId": 255901001
+                      },
+                      "schoolYearTypeReference": {
+                        "schoolYear": 2024
+                      },
+                      "calendarTypeDescriptor": "uri://ed-fi.org/CalendarTypeDescriptor#Student Specific",
```
Comment (bradbanister) [inline src/tests/EdFi.DataManagementService.Tests.E2E/Features/SchoolYearReferenceValidation.feature:37]: Do you have this School and CalendarType already inserted?

File: src/tests/EdFi.DataManagementService.Tests.E2E/Features/SchoolYearReferenceValidation.feature [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                      "currentSchoolYear": true,
+                      "schoolYearDescription": "Current School Year 2024"
+                    }
+                  """
+             Then it should respond with 201 or 200
+              When a POST request is made to "/ed-fi/schoolYearTypes" with
+                  """
+                    {
+                      "schoolYear": 2029,
+                      "currentSchoolYear": true,
```
Comment (bradbanister) [inline src/tests/EdFi.DataManagementService.Tests.E2E/Features/SchoolYearReferenceValidation.feature:20]: Not important to the test, but you should only set one of these currentSchoolYears to true

File: src/tests/EdFi.DataManagementService.Tests.E2E/Features/SchoolYearReferenceValidation.feature [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                      "status": 409,
+                      "correlationId": null
+                     }
+                  """
+
+        #CalendarDates / School Year
+        @Ignore
+        Scenario: Try creating a CalendarDate using a valid Calendar reference
+              
+              When a POST request is made to "/ed-fi/calendarDates" with
```
Comment (bradbanister) [inline src/tests/EdFi.DataManagementService.Tests.E2E/Features/SchoolYearReferenceValidation.feature:75]: This is a Calendar

File: src/tests/EdFi.DataManagementService.Tests.E2E/Features/SchoolYearReferenceValidation.feature [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+             Then it should respond with 201
+
+        @Ignore
+        Scenario: Try creating a CalendarDate using an invalid Calendar reference with an invalid School year
+             When a POST request is made to "/ed-fi/calendarDates" with
+                  """
+                     {
+                        "calendarReference": {
+                          "calendarCode": "2010605675",
+                          "schoolId": 255901001,
```
Comment (bradbanister) [inline src/tests/EdFi.DataManagementService.Tests.E2E/Features/SchoolYearReferenceValidation.feature:115]: Personally, I'd use simpler calendarCodes and schoolIds to make it easier for someone reading the test to see when they are invalid.

File: src/tests/EdFi.DataManagementService.Tests.E2E/Features/SchoolYearReferenceValidation.feature [hunk 5]
```diff
@@ -0,0 +1,138 @@
+Feature: School Year Reference Validation
+  Reference validation on School Years
+
+        Background:
+
+            Given the Data Management Service must receive a token issued by "http://localhost"
+              And user is already authorized
+
+            When a POST request with list of required "descriptors"
```
Comment (stephenfuqua) [inline src/tests/EdFi.DataManagementService.Tests.E2E/Features/SchoolYearReferenceValidation.feature:9]: Usually, we would not have a `When` statement in the `Background`. Should only have `Given` and `And` statements.

File: src/tests/EdFi.DataManagementService.Tests.E2E/Features/SchoolYearReferenceValidation.feature [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+                        "schoolId": 535
+                      },
+                      "schoolYearTypeReference": {
+                        "schoolYear": 2024
+                      },
+                      "calendarTypeDescriptor": "uri://ed-fi.org/CalendarTypeDescriptor#Student Specific",
+                      "gradeLevels": []
+                  }
+                  """
+             Then it should respond with 201 or 200
```
Comment (stephenfuqua) [inline src/tests/EdFi.DataManagementService.Tests.E2E/Features/SchoolYearReferenceValidation.feature:43]: This should be `201`. Receiving `200` would be a bug.

File: src/tests/EdFi.DataManagementService.Tests.E2E/Features/SchoolYearReferenceValidation.feature [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+                      },
+                      "schoolYearTypeReference": {
+                        "schoolYear": 2029
+                      },
+                      "calendarTypeDescriptor": "uri://ed-fi.org/CalendarTypeDescriptor#Student Specific",
+                      "gradeLevels": []
+                  }
+                  """
+             Then it should respond with 201 or 200
+             When a POST request is made to "/ed-fi/calendarDates" with
```
Comment (stephenfuqua) [inline src/tests/EdFi.DataManagementService.Tests.E2E/Features/SchoolYearReferenceValidation.feature:92]: Each scenario should have a single `When` statement. Is the first `When` trying to setup initial date?

Big picture, the scenario could be more descriptive of the _purpose_ for this scenario. We can read to see _what_ it is doing, but we need to understand _why_.

**Review by JBrenesSimpat — APPROVED** [review]

**Pull request merged**